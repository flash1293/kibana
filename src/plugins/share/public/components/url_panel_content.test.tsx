/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

jest.mock('../lib/url_shortener', () => ({}));

import React from 'react';
import { shallow } from 'enzyme';

import { UrlPanelContent } from './url_panel_content';

const URL_PARAM_EXTENSIONS_SELECTOR = '[data-test-subj="urlParamExtensions"]';
const defaultProps = {
  allowShortUrl: true,
  objectType: 'dashboard',
  basePath: '',
  post: () => Promise.resolve({} as any),
};

test('render', () => {
  const component = shallow(<UrlPanelContent {...defaultProps} />);
  expect(component).toMatchSnapshot();
});

test('should enable saved object export option when objectId is provided', () => {
  const component = shallow(<UrlPanelContent {...defaultProps} objectId="id1" />);
  expect(component).toMatchSnapshot();
});

test('should hide short url section when allowShortUrl is false', () => {
  const component = shallow(
    <UrlPanelContent {...defaultProps} allowShortUrl={false} objectId="id1" />
  );
  expect(component).toMatchSnapshot();
});

test('should show url param checkboxes when embedded link', () => {
  const component = shallow(<UrlPanelContent {...defaultProps} isEmbedded={true} objectId="id1" />);
  expect(component.find(URL_PARAM_EXTENSIONS_SELECTOR).length).toBe(1);
  expect(component).toMatchSnapshot();
});

test('should not show url param checkboxes when permalink', () => {
  const component = shallow(<UrlPanelContent {...defaultProps} objectId="id1" />);
  expect(component.find(URL_PARAM_EXTENSIONS_SELECTOR).length).toBe(0);
  expect(component).toMatchSnapshot();
});
