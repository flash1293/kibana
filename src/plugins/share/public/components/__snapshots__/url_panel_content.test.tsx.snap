// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<I18nProvider>
  <EuiForm
    className="kbnShareContextMenu__finalPanel"
    data-test-subj="shareUrlForm"
  >
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      helpText={
        <FormattedMessage
          defaultMessage="Can't share as saved object until the {objectType} has been saved."
          id="share.urlPanel.canNotShareAsSavedObjectHelpText"
          values={
            Object {
              "objectType": "dashboard",
            }
          }
        />
      }
      label={
        <FormattedMessage
          defaultMessage="Generate the link as"
          id="share.urlPanel.generateLinkAsLabel"
          values={Object {}}
        />
      }
      labelType="label"
    >
      <EuiRadioGroup
        idSelected="snapshot"
        onChange={[Function]}
        options={
          Array [
            Object {
              "data-test-subj": "exportAsSnapshot",
              "id": "snapshot",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Snapshot"
                    id="share.urlPanel.snapshotLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself. Edits to the saved {objectType} won't be visible via this URL."
                        id="share.urlPanel.snapshotDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
            Object {
              "data-test-subj": "exportAsSavedObject",
              "disabled": true,
              "id": "savedObject",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Saved object"
                    id="share.urlPanel.savedObjectLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="You can share this URL with people to let them load the most recent saved version of this {objectType}."
                        id="share.urlPanel.savedObjectDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
          ]
        }
      />
    </EuiFormRow>
    <EuiFormRow
      data-test-subj="createShortUrl"
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      labelType="label"
    >
      <EuiFlexGroup
        gutterSize="none"
        responsive={false}
      >
        <EuiFlexItem>
          <EuiSwitch
            checked={false}
            data-test-subj="useShortUrl"
            label={
              <FormattedMessage
                defaultMessage="Short URL"
                id="share.urlPanel.shortUrlLabel"
                values={Object {}}
              />
            }
            onChange={[Function]}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiIconTip
            content={
              <FormattedMessage
                defaultMessage="We recommend sharing shortened snapshot URLs for maximum compatibility. Internet Explorer has URL length restrictions, and some wiki and markup parsers don't do well with the full-length version of the snapshot URL, but the short URL should work great."
                id="share.urlPanel.shortUrlHelpText"
                values={Object {}}
              />
            }
            position="bottom"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiCopy
      afterMessage="Copied"
      anchorClassName="eui-displayBlock"
      textToCopy="http://localhost/"
    >
      <Component />
    </EuiCopy>
  </EuiForm>
</I18nProvider>
`;

exports[`should enable saved object export option when objectId is provided 1`] = `
<I18nProvider>
  <EuiForm
    className="kbnShareContextMenu__finalPanel"
    data-test-subj="shareUrlForm"
  >
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      label={
        <FormattedMessage
          defaultMessage="Generate the link as"
          id="share.urlPanel.generateLinkAsLabel"
          values={Object {}}
        />
      }
      labelType="label"
    >
      <EuiRadioGroup
        idSelected="snapshot"
        onChange={[Function]}
        options={
          Array [
            Object {
              "data-test-subj": "exportAsSnapshot",
              "id": "snapshot",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Snapshot"
                    id="share.urlPanel.snapshotLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself. Edits to the saved {objectType} won't be visible via this URL."
                        id="share.urlPanel.snapshotDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
            Object {
              "data-test-subj": "exportAsSavedObject",
              "disabled": false,
              "id": "savedObject",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Saved object"
                    id="share.urlPanel.savedObjectLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="You can share this URL with people to let them load the most recent saved version of this {objectType}."
                        id="share.urlPanel.savedObjectDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
          ]
        }
      />
    </EuiFormRow>
    <EuiFormRow
      data-test-subj="createShortUrl"
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      labelType="label"
    >
      <EuiFlexGroup
        gutterSize="none"
        responsive={false}
      >
        <EuiFlexItem>
          <EuiSwitch
            checked={false}
            data-test-subj="useShortUrl"
            label={
              <FormattedMessage
                defaultMessage="Short URL"
                id="share.urlPanel.shortUrlLabel"
                values={Object {}}
              />
            }
            onChange={[Function]}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiIconTip
            content={
              <FormattedMessage
                defaultMessage="We recommend sharing shortened snapshot URLs for maximum compatibility. Internet Explorer has URL length restrictions, and some wiki and markup parsers don't do well with the full-length version of the snapshot URL, but the short URL should work great."
                id="share.urlPanel.shortUrlHelpText"
                values={Object {}}
              />
            }
            position="bottom"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiCopy
      afterMessage="Copied"
      anchorClassName="eui-displayBlock"
      textToCopy="http://localhost/"
    >
      <Component />
    </EuiCopy>
  </EuiForm>
</I18nProvider>
`;

exports[`should hide short url section when allowShortUrl is false 1`] = `
<I18nProvider>
  <EuiForm
    className="kbnShareContextMenu__finalPanel"
    data-test-subj="shareUrlForm"
  >
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      label={
        <FormattedMessage
          defaultMessage="Generate the link as"
          id="share.urlPanel.generateLinkAsLabel"
          values={Object {}}
        />
      }
      labelType="label"
    >
      <EuiRadioGroup
        idSelected="snapshot"
        onChange={[Function]}
        options={
          Array [
            Object {
              "data-test-subj": "exportAsSnapshot",
              "id": "snapshot",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Snapshot"
                    id="share.urlPanel.snapshotLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself. Edits to the saved {objectType} won't be visible via this URL."
                        id="share.urlPanel.snapshotDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
            Object {
              "data-test-subj": "exportAsSavedObject",
              "disabled": false,
              "id": "savedObject",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Saved object"
                    id="share.urlPanel.savedObjectLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="You can share this URL with people to let them load the most recent saved version of this {objectType}."
                        id="share.urlPanel.savedObjectDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
          ]
        }
      />
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiCopy
      afterMessage="Copied"
      anchorClassName="eui-displayBlock"
      textToCopy="http://localhost/"
    >
      <Component />
    </EuiCopy>
  </EuiForm>
</I18nProvider>
`;

exports[`should not show url param checkboxes when permalink 1`] = `
<I18nProvider>
  <EuiForm
    className="kbnShareContextMenu__finalPanel"
    data-test-subj="shareUrlForm"
  >
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      label={
        <FormattedMessage
          defaultMessage="Generate the link as"
          id="share.urlPanel.generateLinkAsLabel"
          values={Object {}}
        />
      }
      labelType="label"
    >
      <EuiRadioGroup
        idSelected="snapshot"
        onChange={[Function]}
        options={
          Array [
            Object {
              "data-test-subj": "exportAsSnapshot",
              "id": "snapshot",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Snapshot"
                    id="share.urlPanel.snapshotLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself. Edits to the saved {objectType} won't be visible via this URL."
                        id="share.urlPanel.snapshotDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
            Object {
              "data-test-subj": "exportAsSavedObject",
              "disabled": false,
              "id": "savedObject",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Saved object"
                    id="share.urlPanel.savedObjectLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="You can share this URL with people to let them load the most recent saved version of this {objectType}."
                        id="share.urlPanel.savedObjectDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
          ]
        }
      />
    </EuiFormRow>
    <EuiFormRow
      data-test-subj="createShortUrl"
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      labelType="label"
    >
      <EuiFlexGroup
        gutterSize="none"
        responsive={false}
      >
        <EuiFlexItem>
          <EuiSwitch
            checked={false}
            data-test-subj="useShortUrl"
            label={
              <FormattedMessage
                defaultMessage="Short URL"
                id="share.urlPanel.shortUrlLabel"
                values={Object {}}
              />
            }
            onChange={[Function]}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiIconTip
            content={
              <FormattedMessage
                defaultMessage="We recommend sharing shortened snapshot URLs for maximum compatibility. Internet Explorer has URL length restrictions, and some wiki and markup parsers don't do well with the full-length version of the snapshot URL, but the short URL should work great."
                id="share.urlPanel.shortUrlHelpText"
                values={Object {}}
              />
            }
            position="bottom"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiCopy
      afterMessage="Copied"
      anchorClassName="eui-displayBlock"
      textToCopy="http://localhost/"
    >
      <Component />
    </EuiCopy>
  </EuiForm>
</I18nProvider>
`;

exports[`should show url param checkboxes when embedded link 1`] = `
<I18nProvider>
  <EuiForm
    className="kbnShareContextMenu__finalPanel"
    data-test-subj="shareUrlForm"
  >
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      label={
        <FormattedMessage
          defaultMessage="Generate the link as"
          id="share.urlPanel.generateLinkAsLabel"
          values={Object {}}
        />
      }
      labelType="label"
    >
      <EuiRadioGroup
        idSelected="snapshot"
        onChange={[Function]}
        options={
          Array [
            Object {
              "data-test-subj": "exportAsSnapshot",
              "id": "snapshot",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Snapshot"
                    id="share.urlPanel.snapshotLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself. Edits to the saved {objectType} won't be visible via this URL."
                        id="share.urlPanel.snapshotDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
            Object {
              "data-test-subj": "exportAsSavedObject",
              "disabled": false,
              "id": "savedObject",
              "label": <ForwardRef
                gutterSize="none"
                responsive={false}
              >
                <EuiFlexItem>
                  <FormattedMessage
                    defaultMessage="Saved object"
                    id="share.urlPanel.savedObjectLabel"
                    values={Object {}}
                  />
                </EuiFlexItem>
                <EuiFlexItem
                  grow={false}
                >
                  <EuiIconTip
                    content={
                      <FormattedMessage
                        defaultMessage="You can share this URL with people to let them load the most recent saved version of this {objectType}."
                        id="share.urlPanel.savedObjectDescription"
                        values={
                          Object {
                            "objectType": "dashboard",
                          }
                        }
                      />
                    }
                    position="bottom"
                  />
                </EuiFlexItem>
              </ForwardRef>,
            },
          ]
        }
      />
    </EuiFormRow>
    <EuiFormRow
      data-test-subj="createShortUrl"
      describedByIds={Array []}
      display="row"
      fullWidth={false}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      labelType="label"
    >
      <EuiFlexGroup
        gutterSize="none"
        responsive={false}
      >
        <EuiFlexItem>
          <EuiSwitch
            checked={false}
            data-test-subj="useShortUrl"
            label={
              <FormattedMessage
                defaultMessage="Short URL"
                id="share.urlPanel.shortUrlLabel"
                values={Object {}}
              />
            }
            onChange={[Function]}
          />
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiIconTip
            content={
              <FormattedMessage
                defaultMessage="We recommend sharing shortened snapshot URLs for maximum compatibility. Internet Explorer has URL length restrictions, and some wiki and markup parsers don't do well with the full-length version of the snapshot URL, but the short URL should work great."
                id="share.urlPanel.shortUrlHelpText"
                values={Object {}}
              />
            }
            position="bottom"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFormRow>
    <EuiHorizontalRule />
    <EuiCheckboxGroup
      data-test-subj="urlParamExtensions"
      idToSelectedMap={
        Object {
          "filterBar": true,
          "query": false,
          "timeFilter": false,
          "topMenu": false,
        }
      }
      legend={
        Object {
          "children": <FormattedMessage
            defaultMessage="Include"
            id="share.urlPanel.include"
            values={Object {}}
          />,
        }
      }
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "topMenu",
            "label": <FormattedMessage
              defaultMessage="Top menu"
              id="share.urlPanel.topMenu"
              values={Object {}}
            />,
          },
          Object {
            "id": "query",
            "label": <FormattedMessage
              defaultMessage="Query"
              id="share.urlPanel.query"
              values={Object {}}
            />,
          },
          Object {
            "id": "timeFilter",
            "label": <FormattedMessage
              defaultMessage="Time filter"
              id="share.urlPanel.timeFilter"
              values={Object {}}
            />,
          },
          Object {
            "id": "filterBar",
            "label": <FormattedMessage
              defaultMessage="Filter bar"
              id="share.urlPanel.filterBar"
              values={Object {}}
            />,
          },
        ]
      }
    />
    <EuiSpacer
      size="m"
    />
    <EuiCopy
      afterMessage="Copied"
      anchorClassName="eui-displayBlock"
      textToCopy="<iframe src=\\"http://localhost/?embed=true\\" height=\\"600\\" width=\\"800\\"></iframe>"
    >
      <Component />
    </EuiCopy>
  </EuiForm>
</I18nProvider>
`;
